File: src/main/java/jk/kamoru/flayground/info/domain/Tag.java
package jk.kamoru.flayground.info.domain;

import java.util.Date;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class Tag implements Info<Integer> {

  @NotBlank Integer id;
  @NotBlank String name;
  String description;
  long lastModified;

  public Tag(Integer key) {
    setKey(key);
    this.name = "";
    this.description = "";
    this.lastModified = -1;
  }

  @Override
  public Integer getKey() {
    return id;
  }

  @Override
  public void setKey(Integer key) {
    this.id = key;
  }

  @Override
  public void touch() {
    lastModified = new Date().getTime();
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj)
      return true;
    if (obj == null)
      return false;
    if (getClass() != obj.getClass())
      return false;
    Tag other = (Tag) obj;
    if (id == null) {
      if (other.id != null)
        return false;
    } else if (!id.equals(other.id))
      return false;
    return true;
  }

  @Override
  public int hashCode() {
    final int prime = 31;
    int result = 1;
    result = prime * result + ((id == null) ? 0 : id.hashCode());
    return result;
  }

}

File: web-backend/src/domain/tag.ts
/** Tag 도메인 (Info<Integer> 대응) */
export interface Tag {
  id: number;
  name: string;
  group: string;
  description: string;
  lastModified: number;
  /** 조회 시 집계된 사용 횟수 (선택적) */
  count?: number;
}

/**
 * 새 Tag 생성
 * @param id Tag ID
 */
export function createTag(id: number): Tag {
  return { id, name: '', group: '', description: '', lastModified: -1 };
}
